### PR TITLE
Clarify the "enter(-ed)" events

### PR DESCRIPTION
**Pull request by thePanz**
Added clarification about the marker-store for the `enter` and `entered` events.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@lyrixx since you are our biggest Workflow expert, could you please review these proposed changes? Thanks!

File: workflow/usage.rst [hunk 1]
```diff
@@ -255,8 +255,9 @@ order:
     * ``workflow.[workflow name].transition.[transition name]``
 
 ``workflow.enter``
-    The object entered a new place. This is the first event where the object
-    is marked as being in the new place.
+    The object is about to enter a new place. This is the event where the object is
+    going to be marked as being in the new place.
+    Please notice: the marker store of the object is not yet updated with the new state.
```
Comment (lyrixx) [inline workflow/usage.rst]: Nice catch. But the vocabulary is not accurate.
> Please notice : the marking of the object is not ...

The `marking` is the representation of the subject in the workflow
The `marking store` is the interface between the subject and the workflow
The `marker` does not exist

s/state/places/ (plural!)

File: workflow/usage.rst [hunk 2]
```diff
@@ -265,8 +266,8 @@ order:
     * ``workflow.[workflow name].enter.[place name]``
 
 ``workflow.entered``
-    Similar to ``workflow.enter``, except the marking store is updated before this
-    event (making it a good place to flush data in Doctrine).
+    The object has entered a state and the marker store is updated (making it a good
```
Comment (lyrixx) [inline workflow/usage.rst]: * has entered in the place (I'm not so fluent, but we should talk about place and not state)
* `and the marking store`

File: workflow/usage.rst [hunk 3]
```diff
@@ -255,8 +255,9 @@ order:
     * ``workflow.[workflow name].transition.[transition name]``
 
 ``workflow.enter``
-    The object entered a new place. This is the first event where the object
-    is marked as being in the new place.
+    The object is about to enter a new place. This is the event where the object is
+    going to be marked as being in the new place.
```
Comment (lyrixx) [inline workflow/usage.rst]: Actually, this is the event **before** the subject (everywhere in the code I used subject and not object, I think it's better if the doc reflects that) is going to be updated

**Review by lyrixx — CHANGES_REQUESTED** [review]
Thanks for the clarification.

I let few minor comments

File: workflow/usage.rst (hunk 2 above)
Comment (thePanz) [reply workflow/usage.rst]: Fixed!

File: workflow/usage.rst (hunk 3 above)
Comment (thePanz) [reply workflow/usage.rst]: Updated as:
```
    The subject is about to enter a new place. This is the event triggered before the subject is
    going to be updated as being in the new places.
    Please notice: the marking of the subject is not yet updated with the new places.
```

**Comment by thePanz** [comment]
Thanks @javiereguiluz  and @lyrixx for the review!
I updated the PR following your comments!

**Review by lyrixx — APPROVED** [review]
:+1: for the technical part.
I let others review the style ;)

Thanks for this PR.

**Note:** The target branch (4.0) is wrong, it should be 3.3

**Comment by thePanz** [comment]
Thanks @lyrixx I changed the target branch and rebased my changes on the `3.3` branch.

**Comment by javiereguiluz** [comment]
@thePanz thanks for this contribution and for your patience during the review process. Regarding the right branch for this change, it should be 3.4 because 3.3 is no longer maintained. For future contributions, don't worry much about the branch because doc maintainers can change it when merging. Thanks!

**Comment by thePanz** [comment]
Thanks @javiereguiluz ! :)

**Pull request closed without merging**